roles/mining-proxy/src/main.rs

```diff
@@ -137,5 +138,18 @@ async fn main() {
     );
 
     info!("PROXY INITIALIZED");
-    crate::lib::downstream_mining::listen_for_downstream_mining(socket).await
+
+    let (shutdown_tx, shutdown_rx) = oneshot::channel();
+
+    tokio::select! {
+        _ = lib::downstream_mining::listen_for_downstream_mining(socket, shutdown_rx) => {
+            warn!("Downstream mining listener exited prematurely");
+        },
+        _ = tokio::signal::ctrl_c() => {
```

Comment:
Could we move this internally to `listen_for_downstream_minining` and avoid nesting `tokio::select!`?

Reply:
We could, but it seemed to me better to leave the `main` fn handling signal and other executable related stuff while leaving `listen_for_downstream_minining` cancellable but only focused on accepting connections.
Anyway, if you feel strongly about this, I'll change :)